shuup/campaigns/models/campaigns.py

```diff
@@ -117,6 +119,13 @@ def rules_match(self, context, shop_product, matching_catalog_filters, matching_
         return True
 
     @classmethod
+    def get_for_product(cls, shop_product, limit=100):
+        matching_filters = get_matching_for_product(shop_product, provide_category="campaign_catalog_filter")
+        matching_conditions = get_matching_for_product(shop_product, provide_category="campaign_context_condition")
+        query_filter = Q(Q(filters__in=matching_filters) | Q(conditions__in=matching_conditions))
+        return cls.objects.filter(shop=shop_product.shop, active=True).filter(query_filter).distinct()[:limit]
```

Comment:
I think we should not limit these at the moment. I will remove it. I think the proper way in future would be a pagination of sorts.

I created the queryset manager and it will be there in next pr.